Sources/PromotedFirebaseRemoteConfig/FirebaseRemoteConfigService.swift

```diff
@@ -8,17 +8,63 @@ import PromotedCore
 
 final class FirebaseRemoteConfigConnection: RemoteConfigConnection {
 
-  private let firebaseApp: FirebaseApp
+  private let plistFilename: String?
+  private var cachedFirebaseApp: FirebaseApp?
 
   init(app: FirebaseApp) {
-    self.firebaseApp = app
+    self.plistFilename = nil
+    self.cachedFirebaseApp = app
+  }
+
+  init(plistFilename: String) {
+    self.plistFilename = plistFilename
+    self.cachedFirebaseApp = nil
+  }
+
+  private func firebaseApp() throws -> FirebaseApp {
+    // Use existing cached app, if available.
+    if let existingApp = cachedFirebaseApp { return existingApp }
+
+    // If the app has already been configured in Firebase, re-use it.
```

Comment:
Side-comment - is there an easy way for clients to disable this in case they are concerned about a Firebase dependency?

Reply:
Do you mean via remote config? No, we don't allow the remote config configuration to be changed remotely. This prevents clients from being locked out of remote updates.

Locally, it's pretty easy to add/remove this dependency. You change 3 lines in the code configuration. To enable remote config, use:
```
let moduleConfig = ModuleConfig.defaultConfig()
moduleConfig.useFirebaseRemoteConfig(plistFilename: "GoogleService-Info-Promoted")
var clientConfig = ClientConfig()
clientConfig.metricLoggingURL = "..."
moduleConfig.initialConfig = clientConfig
let service = try MetricsLoggerService(moduleConfig: moduleConfig)
```
To remove remote config, remove the `moduleConfig.useFirebaseRemoteConfig()` call, or use `ClientConfig` directly:
```
var clientConfig = ClientConfig()
clientConfig.metricLoggingURL = "..."
let service = try MetricsLoggerService(initialConfig: clientConfig)
```
In addition to this code change, clients would need to make the following build dependency changes.
1. If they use CocoaPods, add the `pod 'PromotedAIMetricsSDK/FirebaseRemoteConfig'` line from their Podfile.
2. If they use Swift Package Manager, add the dependency `PromotedFirebaseRemoteConfig` from our package.